### PR TITLE
catatoinit: rewrite LISTEN_PID to the child process

### PR DESCRIPTION
**Pull request by giuseppe**
In order to get systemd fd passing let's rewrite the LISTEN_PID
environment variable if it is pointing to catatonit init process.

Closes: https://github.com/openSUSE/catatonit/issues/13

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by alexlarsson** [comment]
Didn't test it, but lgtm

File: catatonit.c [hunk 1]
```diff
@@ -193,20 +194,46 @@ static int make_foreground(sigset_t *sigmask)
...
 {
+	pid_t parent_pid = getpid();
 	pid_t child = fork();
 	if (child != 0) {
 		if (child < 0)
 			error("failed to fork child: %m");
 		return child;
 	}
 
+        rewrite_listen_pid_env(parent_pid);
```
Comment (cyphar) [inline catatonit.c:235]: Indentation is messed for some reason?

File: catatonit.c [hunk 2]
```diff
@@ -193,20 +194,46 @@ static int make_foreground(sigset_t *sigmask)
...
+}
+
 /*
  * Spawn a child process with the given arguments and signal map and make it a
  * faux-pid1 by placing it in the foreground. This is the main process which
  * catatonit is going to be managing throughout its life.
  */
 static int spawn_pid1(char *file, char **argv, sigset_t *sigmask)
 {
+	pid_t parent_pid = getpid();
```
Comment (cyphar) [inline catatonit.c:227]: `getppid()` is simpler and doesn't require saving a variable. So `rewrite_listen_pid_env` doesn't really need to take an argument.

File: catatonit.c [hunk 3]
```diff
@@ -193,20 +194,46 @@ static int make_foreground(sigset_t *sigmask)
...
+		int r;
+
+		sprintf(pid_str, "%d", getpid());
+
+		r = setenv("LISTEN_PID", pid_str, 1);
+		if (r < 0)
+			warn("could not overwrite env variable LISTEN_PID: %m");
+	}
+}
+
```
Comment (cyphar) [inline catatonit.c:219]: While I get that on paper it's safe in this specific case, I don't like the use of `sprintf` so please use `snprintf` or `asprintf` -- but more importantly I'd prefer `strtoll` for the pid equivalence check.

**Review by cyphar — CHANGES_REQUESTED** [review]
As with the other PR, I didn't see this until now. Sorry about that.

**Comment by giuseppe** [comment]
thanks for the review.  I've addressed your comments in the last commit

**Review by cyphar — APPROVED** [review]
LGTM.

**Comment by cyphar** [comment]
I'll do a new release once #14 is merged.

**Pull request merged**